frontend/test/pytest/test_conditionals.py

```diff
@@ -231,6 +231,22 @@ def arithi(x, y, op1, op2):
 
         assert arithi(x, y, op1, op2) == arithc(x, y, op1, op2)
 
+    def test_no_true_false_parameters(self):
+        """Test non-empty parameter detection in conditionals"""
+        with pytest.raises(TypeError):
+
+            @qjit
+            def arithc():
+                @cond(True)
+                def branch(_):
+                    return 1
+
+                @branch.otherwise
+                def branch(_):
+                    return 0
```

Comment:
I think it would also be good to separate these under two `with pytest.raises` so that we can be sure that the message for works for both branches.

Reply:
Good point, thanks!